### ISSUE TITLE
Segmentation fault on linux with exec mode

### ISSUE THREAD

**Issue by abeaumont** [pyroscope-io/pyroscope#503]
I get a segmentation fault running the latest [linux binary](https://dl.pyroscope.io/release/pyroscope-0.2.4-linux-amd64.tar.gz) using exec mode:

```
./pyroscope exec python dummy.py
INFO[2021-11-05T09:33:08.670679] exec/cli.go:87 to disable logging from pyroscope, specify -no-logging flag
INFO[2021-11-05T09:33:08.670928] exec/cli.go:91 we recommend specifying application name via -application-name flag or env variable PYROSCOPE_APPLICATION_NAME
INFO[2021-11-05T09:33:08.671034] exec/cli.go:94 for now we chose the name for you and it's "pyspy.nostalgic.poincare"
Segmentation fault (core dumped)
```

The backtrace:

```
➤  bt
#0  0x0000000001052157 in __rust_probestack ()
#1  0x0000000000d26944 in py_spy::python_spy::check_interpreter_addresses ()
#2  0x0000000000d09323 in py_spy::python_spy::PythonSpy::new ()
#3  0x0000000000cc3fd2 in pyspy_init ()
#4  0x0000000000a89fe0 in _cgo_04a990bedb22_Cfunc_pyspy_init (v=0xc00014b198) at cgo-gcc-prolog:80
#5  0x000000000047fb50 in runtime.asmcgocall () at runtime/asm_amd64.s:765
#6  0x00007fffd1140930 in ?? ()
#7  0x0000000000455ccd in runtime.park_m (gp=0xed0) at runtime/proc.go:3516
#8  0x000000000047dc23 in runtime.mcall () at runtime/asm_amd64.s:307
#9  0x0000000000000000 in ?? ()
```

**Comment by petethepig** [comment]
Can you provide some more info about the environment as well as the source code for the script?

I'm particularly interested in linux version, go version, rust version and python version

**Comment by abeaumont** [comment]
Sure. I used the published binary, which is statically linked, so the go and rust versions shouldn't be relevant, but I'll provide them anyway.

Linux version: `Linux 5.14.15-arch1-1 #1 SMP PREEMPT Wed, 27 Oct 2021 23:38:14 +0000 x86_64`
Go version: `go version go1.17.2 linux/amd64`
Rust version: rustc 1.56.1 (59eed8a2a 2021-11-01)
Python version: Python 3.9.7
The script is something minimal, as it seems to crash with anything I tried. `dummy.py` is:
```python
from time import sleep

sleep(5)
```
